frontend/ui/network/manager.lua

```diff
@@ -60,6 +61,39 @@ function NetworkMgr:promptWifiOff(complete_callback)
     })
 end
 
+function NetworkMgr:turnOnWifiAndWaitForConnection(callback)
+    NetworkMgr:turnOnWifi()
+    local timeout = 60;
+    local retry_count = 0;
+    local info = InfoMessage:new{ text = T(_("Enabling Wi-Fi. Waiting for Internet connection…\nTimeout %1 seconds."), timeout)}
+    UIManager:show(info)
+    UIManager:forceRePaint()
+    while not NetworkMgr:isOnline() and retry_count < timeout do
+        FFIUtil.sleep(1)
+        retry_count = retry_count + 1
+    end
+    UIManager:close(info)
+    if retry_count == timeout then
+        UIManager:show(InfoMessage:new{ text = _("Error obtaining Internet connection. Operation aborted...") })
```

Comment:
How about simply `Error connecting to the network.`?

Reply:
ok